cmd/algokey/part.go

```diff
@@ -0,0 +1,175 @@
+	},
+}
+
+var partGenerateCmd = &cobra.Command{
+	Use:   "generate",
+	Short: "Generate participation key",
+	Args:  cobra.NoArgs,
+	Run: func(cmd *cobra.Command, _ []string) {
+		if partLastRound < partFirstRound {
+			fmt.Fprintf(os.Stderr, "Last round %d < first round %d\n", partLastRound, partFirstRound)
+			os.Exit(1)
+		}
+
+		if partKeyDilution == 0 {
+			partKeyDilution = 1 + uint64(math.Sqrt(float64(partLastRound-partFirstRound)))
+		}
+
+		var err error
+		var parent basics.Address
+		if partParent != "" {
+			parent, err = basics.UnmarshalChecksumAddress(partParent)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "Cannot parse parent address %s: %v\n", partParent, err)
+				os.Exit(1)
+			}
+		}
+
+		partdb, err := db.MakeErasableAccessor(partKeyfile)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Cannot open partkey database %s: %v\n", partKeyfile, err)
+			os.Exit(1)
+		}
+
+		partkey, err := account.FillDBWithParticipationKeys(partdb, parent, basics.Round(partFirstRound), basics.Round(partLastRound), partKeyDilution)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Cannot generate partkey database %s: %v\n", partKeyfile, err)
+			os.Exit(1)
+		}
+
+		printPartkey(partkey)
+	},
+}
+
+var partInfoCmd = &cobra.Command{
+	Use:   "info",
+	Short: "Print participation key information",
+	Args:  cobra.NoArgs,
+	Run: func(cmd *cobra.Command, _ []string) {
+		partdb, err := db.MakeErasableAccessor(partKeyfile)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Cannot open partkey database %s: %v\n", partKeyfile, err)
+			os.Exit(1)
+		}
+
+		partkey, err := account.RestoreParticipation(partdb)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Cannot load partkey database %s: %v\n", partKeyfile, err)
+			os.Exit(1)
+		}
+
+		printPartkey(partkey)
+	},
+}
+
+var partReparentCmd = &cobra.Command{
+	Use:   "reparent",
+	Short: "Change parent address of participation key",
+	Args:  cobra.NoArgs,
+	Run: func(cmd *cobra.Command, _ []string) {
+		parent, err := basics.UnmarshalChecksumAddress(partParent)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Cannot parse parent address %s: %v\n", partParent, err)
+			os.Exit(1)
+		}
+
+		partdb, err := db.MakeErasableAccessor(partKeyfile)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Cannot open partkey database %s: %v\n", partKeyfile, err)
+			os.Exit(1)
+		}
+
+		partkey, err := account.RestoreParticipation(partdb)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Cannot load partkey database %s: %v\n", partKeyfile, err)
+			os.Exit(1)
+		}
+
+		partkey.Parent = parent
+		err = partkey.PersistNewParent()
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Cannot persist partkey database %s: %v\n", partKeyfile, err)
+			os.Exit(1)
+		}
+
+		printPartkey(partkey)
+	},
+}
+
+func printPartkey(partkey account.Participation) {
+	fmt.Printf("Parent address:    %s\n", partkey.Parent.GetChecksumAddress().String())
+	fmt.Printf("VRF public key:    %s\n", base64.StdEncoding.EncodeToString(partkey.VRF.PK[:]))
```

Comment:
Are these (VRF public key, Voting public key) the same encodings as seen in genesis.json and the `goal account partkeyinfo` command?